roles/kube_prometheus_stack/defaults/main.yml

```diff
@@ -12,46 +12,158 @@ kube_prometheus_release_name: kube-prometheus-stack
+  }}
+
+# Size of the PVC for the prometheus database
+kube_prometheus_stack_prometheus_spec_volume_gb: 10
+
+# Max size of the prometheus database
+kube_prometheus_stack_prometheus_spec_retention_size: "{{ ( 0.95 * kube_prometheus_stack_prometheus_spec_volume_gb ) ~ 'GB' }}"
+
+# Spec for Prometheus pods
+kube_prometheus_stack_prometheus_spec: >-
+  {{-
+    {
+      "retentionSize": kube_prometheus_stack_prometheus_spec_retention_size,
+      "storageSpec": {
+        "volumeClaimTemplate": {
+          "spec": {
+            "accessModes": ["ReadWriteOnce"],
+            "resources": {
+              "requests": {
+                "storage": kube_prometheus_stack_prometheus_spec_volume_gb ~ 'Gi'
+                }
+              }
+            }
+          }
+        }
+      } 
+    }}
+
+# Default values for Prometheus section of kube-prometheus-stack
+kube_prometheus_stack_prometheus_values_defaults: >-
+  {{-
+    {
+    "prometheusSpec": kube_prometheus_stack_prometheus_spec
+    }
+  }}
+
+# Overrides for Prometheus section of kube-prometheus-stack
+kube_prometheus_stack_prometheus_values_overrides: {}
+
+# Values to pass to Prometheus part of kube-prometheus-stack
+kube_prometheus_stack_prometheus_values: >-
+  {{-
+      kube_prometheus_stack_prometheus_values_defaults |
+        combine(
+          kube_prometheus_stack_prometheus_values_overrides,
+          recursive = True
+        )
+  }}
+
+# List of namespaces in which to look for ConfigMaps with the label
+# `grafana_dashboard: "1"` which signifies that the map contains
+# additional dashboards to be imported into Grafana
+kube_prometheus_stack_grafana_sidecar_dashboard_search_ns:
+  - monitoring-system
+  - azimuth
+
+# Values to pass to Grafana part of kube-prometheus-stack
+kube_prometheus_stack_grafana_values: >-
+  {{-
+    {
+      "sidecar": {
+        "dashboards": {
+          "searchNamespace": kube_prometheus_stack_grafana_sidecar_dashboard_search_ns
+        }
+      }
+    }
+  }}
+
+# Values to pass to Alertmanager part of kube-prometheus-stack
+kube_prometheus_stack_alertmanager_values: >-
+  {{-
+    {
+      "enabled": false,
+      "config": {
+        "receivers": {
+          "name": "null"
+          }
+        }
+      } | 
+        combine(
+          {
+            "enabled": azimuth_management_cluster_alerting_enabled
+            }
+            if azimuth_management_cluster_alerting_enabled
+            else {}
+          ) |
+            combine(
+              {     
+                "config": {
+                  "global": {
+                    "slack_api_url": kube_prometheus_stack_alertmanager_slack_url
+                  },
+                  "route": {
+                    "receiver": 'slack-notifications',
+                    "group_by": [
+                      "namespace"
+                    ]
+                  },
+                  "receivers": [
+                      {
+                        "name": "null"
+                      },
```

Comment:
Do we need to define the null receiver again here if it's already defined above and we're using combine? Or is this because we're not using combine(..., recursive = True)?

Reply:
I didn't look into this behaviour too deeply, but good spot. I think we'd need to do something with the `list_merge` parameter for `combine()`: https://docs.ansible.com/ansible/devel/collections/ansible/builtin/combine_filter.html#parameter-list_merge

Reply:
Fixed and tested